packages/core/src/config/load.ts

```diff
@@ -61,6 +48,55 @@ export async function loadConfig(configPath?: string, customExtends?: string[]):
   );
 }
 
+function transformApiDefinitionsToApis(apiDefinitions: Record<string, string> = {}): Record<string, Api> {
+  const apis: Record<string, Api> = {};
+  for (const [apiName, apiPath] of Object.entries(apiDefinitions)) {
+    apis[apiName] = { root: apiPath };
+  }
+  return apis;
+}
+
+function decorateWithLatest(rawConfig: RawConfig): RawConfig {
+  let apis: Record<string, Api> = {};
+  for (const [key, value] of Object.entries(rawConfig.apis || {})) {
+    const [name, version] = key.split('@');
+    apis[`${name}@${version || 'latest'}`] = value;
+  }
+  return { ...rawConfig, apis };
+}
+
+function transformConfig(rawConfig: DeprecatedRawConfig | RawConfig): RawConfig {
+  if ((rawConfig as RawConfig).apis && (rawConfig as DeprecatedRawConfig).apiDefinitions ||
+    (rawConfig as RawConfig)['features.openapi'] && (rawConfig as DeprecatedRawConfig).referenceDocs
+  ) {
+    throw new Error('Do not use old & new config syntax simultaneously');
+  }
+  const { apiDefinitions, referenceDocs, ...rest } = rawConfig as DeprecatedRawConfig & RawConfig;
+  if (apiDefinitions || referenceDocs) {
+    // TODO: add link to docs.
+    // TODO: show warning without throwing error.
+    throw new Error('apiDefinitions & referenceDocs fields are deprecated. Use apis & features.openapi instead (see  docs: /.//) ');
```

Comment:
Please put a meaningful comment. 
@adamaltman, @RomanHotsiy  could you suggest one?

Reply:
Well, I think this doesn't make sense:

```
  if (apiDefinitions || referenceDocs) {
```

Each should have its own warning (not error). The warning message can be:

- The 'apiDefinitions' field is deprecated. Use 'apis' instead.
- The 'referenceDocs' field is deprecated. Use 'features.openapi' instead.

Reply:
@adamaltman thx

Reply:
done